File: 20A40/cir-agus/EJER_23/MonedaPy/Moneda.py
class Moneda:        
    def __init__(self, divisa, valor, pais, año, escudo):
        self.__divisa = divisa
        self.__valor = valor
        self.__pais = pais
        self.__año = año
        self.__escudo = escudo

    def setDivisa(self, divisa) -> bool:
        if len(divisa) > 0:
            self.__divisa = divisa
            return True
        else: return False
    
    def setValor(self, valor) -> bool:
        if valor > 0:
            self.__valor = valor
            return True
        else: return False
    
    def setPais(self, pais) -> bool:
        if len(pais) > 0:
            self.__pais = pais
            return True
        else: return False
    
    def setAño(self, año) -> bool:
        if año > 0:
            self.__año = año
            return True
        else: return False

    def setEscudo(self, escudo: str) -> bool:
        if len(escudo) > 0:
            self.__escudo = escudo
            return True
        else: return False

    def getDivisa(self) -> str: return self.__divisa
    def getValor(self) -> float: return self.__valor
    def getPais(self) -> str: return self.__pais
    def getAño(self) -> int: return self.__año
    def getEscudo(self) -> str: return self.__escudo

    def toString(self) -> str:
        cadena = f"Divisa: {self.__divisa}\n"
        cadena += f"Valor: {self.__valor}\n"
        cadena += f"Pais: {self.__pais}\n"
        cadena += f"Año: {self.__año}\n"
        cadena += f"Escudo: {self.__escudo}\n"
        return cadena

File: 41A60/cir-agus/EJER_45-Java/Moneda.java
public class Moneda{
    private String Divisa;
    private double Valor;
    private String Pais;
    private int Año;
    private String Escudo;

    public Moneda(){}
    
    public Moneda(String div,double val, String pai, int año,String esc){
        this.Divisa = div;
        this.Valor = val;
        this.Pais = pai;
        this.Año = año;
        this.Escudo = esc;
    
    }

    public boolean setDivisa(String div){
        if(div.length() > 0){
            this.Divisa = div;
            return true;
        }else return false;
    }

    public boolean setValor(double val){
        if(val > 0){
            this.Valor = val;
            return true;
        } else return false;
    }

    public boolean setPais(String pai){
        if(pai.length() > 0){
            this.Pais = pai;
            return true;
        } else return false;
    }

    public boolean setAño(int año){
        if( año > 0){
            this.Año = año;
            return true;
        }else return false;
    }

    public boolean setEscudo(String esc){
        if(esc.length() > 0){
            this.Escudo = esc;
            return true;
        }else return false;
    }

    public String getDivisa(){ return this.Divisa; }
    public double getValor(){ return this.Valor; }
    public String getPais(){ return this.Pais; }
    public int getAño(){ return this.Año; }
    public String getEscudo(){ return this.Escudo; }

    public String toString(){
        String valor = String.valueOf(this.Valor);
        String year = String.valueOf(this.Año);
        
        String cadena = "----------------------------\n";
        cadena += "La divisa: "+this.Divisa+"\n";
        cadena += " El valor: "+valor+"\n";
        cadena += " El pais: "+this.Pais+"\n";
        cadena += " El año: "+year+" \n";
        cadena += " El escudo: "+this.Escudo+"\n";
        cadena += "----------------------------\n";
        return cadena;
    }
}